src/main/client_data.cpp

```diff
@@ -114,7 +116,12 @@ class ClientBufferManager : public BufferManager {
 		return buffer_manager.GetBlockSize();
 	}
 	idx_t GetQueryMaxMemory() const override {
-		return buffer_manager.GetQueryMaxMemory();
+		idx_t global_budget = buffer_manager.GetQueryMaxMemory();
+		const auto &config = ClientConfig::GetConfig(context);
+		if (config.query_memory_limit == DConstants::INVALID_INDEX) {
```

Comment:
We prefer to use `optional_idx` for indices where `DConstants::INVALID_INDEX` means that it has not been set